server/src/parser.rs

```diff
@@ -0,0 +1,698 @@
+        )
+        .is_err());
+    }
+
+    #[test]
+    fn test_tokenize_and_parse_create_model_table_without_table_name() {
+        assert!(tokenize_and_parse_sql(
+            "CREATE MODEL TABLE(timestamp TIMESTAMP, field FIELD, tag TAG)",
+        )
+        .is_err());
+    }
+
+    #[test]
+    fn test_tokenize_and_parse_create_model_table_without_table_table_name_space() {
+        assert!(tokenize_and_parse_sql(
+            "CREATE MODEL TABLEtable_name(timestamp TIMESTAMP, field FIELD, tag TAG)",
+        )
+        .is_err());
+    }
+
+    #[test]
+    fn test_tokenize_and_parse_create_model_table_without_start_parentheses() {
+        assert!(tokenize_and_parse_sql(
+            "CREATE MODEL TABLE table_name timestamp TIMESTAMP, field FIELD, tag TAG)",
+        )
+        .is_err());
+    }
+
+    #[test]
+    fn test_tokenize_and_parse_create_model_table_with_option() {
+        assert!(tokenize_and_parse_sql(
+            "CREATE MODEL TABLE table_name(timestamp TIMESTAMP PRIMARY KEY, field FIELD, tag TAG)",
+        )
+        .is_err());
+    }
+
+    #[test]
+    fn test_tokenize_and_parse_create_model_table_with_sql_types() {
+        assert!(tokenize_and_parse_sql(
+            "CREATE MODEL TABLE table_name(timestamp TIMESTAMP, field REAL, tag VARCHAR)",
+        )
+        .is_err());
+    }
+
+    #[test]
+    fn test_tokenize_and_parse_create_model_table_without_column_name() {
+        assert!(tokenize_and_parse_sql(
+            "CREATE MODEL TABLE table_name(TIMESTAMP, field FIELD, tag TAG)",
+        )
+        .is_err());
+    }
+
+    #[test]
+    fn test_tokenize_and_parse_create_model_table_without_column_type() {
+        assert!(tokenize_and_parse_sql(
+            "CREATE MODEL TABLE table_name(timestamp, field FIELD, tag TAG)",
+        )
+        .is_err());
+    }
+
+    #[test]
+    fn test_tokenize_and_parse_create_model_table_without_comma() {
+        assert!(tokenize_and_parse_sql(
+            "CREATE MODEL TABLE table_name(timestamp TIMESTAMP field FIELD, tag TAG)",
+        )
+        .is_err());
+    }
+
+    #[test]
+    fn test_tokenize_and_parse_create_model_table_without_end_parentheses() {
+        assert!(tokenize_and_parse_sql(
+            "CREATE MODEL TABLE table_name(timestamp TIMESTAMP, field FIELD, tag TAG",
+        )
+        .is_err());
+    }
+
+    #[test]
+    fn test_tokenize_and_parse_two_create_model_table_commands() {
+        let error = tokenize_and_parse_sql(
+            "CREATE MODEL TABLE table_name(timestamp TIMESTAMP, field FIELD, tag TAG);
+             CREATE MODEL TABLE table_name(timestamp TIMESTAMP, field FIELD, tag TAG)",
+        );
+
+        assert!(error.is_err());
+
+        let expected_error =
+            ParserError::ParserError("Multiple SQL commands are not supported.".to_owned());
+        assert!(error.unwrap_err() == expected_error);
+    }
+
+    fn new_object_name(name: &str) -> ObjectName {
```

Comment:
Maybe move this and `new_column_option_def_error_bound()` up under the test function where they are used.